fe/fe-core/src/main/java/org/apache/doris/datasource/hudi/HudiSchemaCacheValue.java

```diff
@@ -20,14 +20,30 @@
 import org.apache.doris.catalog.Column;
 import org.apache.doris.datasource.hive.HMSSchemaCacheValue;
 
+import org.apache.hudi.common.table.HoodieTableMetaClient;
+import org.apache.hudi.common.util.InternalSchemaCache;
+import org.apache.hudi.internal.schema.InternalSchema;
+import org.apache.hudi.internal.schema.utils.InternalSchemaUtils;
+
 import java.util.List;
+import java.util.TreeMap;
 
 public class HudiSchemaCacheValue extends HMSSchemaCacheValue {
 
     private List<String> colTypes;
 
-    public HudiSchemaCacheValue(List<Column> schema, List<Column> partitionColumns) {
+    boolean enableSchemaEvolution;
+
+    TreeMap<Long, InternalSchema> historySchemaCache;
+    // schema version id (timestamp) =>  hudi internal schema.
+
+    public HudiSchemaCacheValue(List<Column> schema, List<Column> partitionColumns,
+            HoodieTableMetaClient hudiClient, boolean enableSchemaEvolution) {
         super(schema, partitionColumns);
+        this.enableSchemaEvolution = enableSchemaEvolution;
+        if (enableSchemaEvolution) {
+            historySchemaCache = InternalSchemaCache.getHistoricalSchemas(hudiClient);
```

Comment:
I support to get this `historySchemaCache` outside the HudiSchemaCacheValue's contructor.
Let this cache value simple